sync/src/test/java/tech/pegasys/artemis/sync/FetchRecentBlocksServiceTest.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.sync;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.artemis.datastructures.util.DataStructureUtil;
+import tech.pegasys.artemis.networking.eth2.Eth2Network;
+import tech.pegasys.artemis.sync.FetchBlockTask.FetchBlockResult;
+import tech.pegasys.artemis.sync.FetchBlockTask.FetchBlockResult.Status;
+import tech.pegasys.artemis.sync.FetchRecentBlocksService.FetchBlockTaskFactory;
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.util.async.StubAsyncRunner;
+
+@ExtendWith(MockitoExtension.class)
+public class FetchRecentBlocksServiceTest {
+
+  @Mock private Eth2Network eth2Network;
```

Comment:
Why use the @Mock annotation instead of the `mock()` method?

Reply:
I'm mainly using it because I don't know create a generic mock (`PendingPool<SignedBeaconBlock>`) with the `mock()` method without using casting and suppressing warnings.